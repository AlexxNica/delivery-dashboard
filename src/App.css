/* Delivery Dashboard custom styles */
.well {
  padding: 15px;
}

.well p {
  margin-top: 10px;
}

.highlight {
  background-color: lightblue;
  color: #000;
  padding: .2em .3em 0;
  border-radius: 2px;
}

.loader {
    border: 16px solid #999;
    border-top: 16px solid #666;
    border-radius: 50%;
    width: 120px;
    height: 120px;
    animation: spin 2s linear infinite;
    margin: 50px auto;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}

/* Releases status list table */
td {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 65px;
  text-align: center;
  padding-bottom: 50px !important;
}

/* Search form */
.search-form .form-group {
  margin-bottom: 0;
}

.clearable-text {
  width: 100%;
}
.clearable-text input {
  width: 100%;
  padding-right: 18px;
}

.text-clear-btn {
  position: absolute;
  right: 5px;
  top: 0;
  bottom: 0;
  height: 14px;
  margin: auto;
  font-size: 14px;
  cursor: pointer;
  color: #ccc;
}
