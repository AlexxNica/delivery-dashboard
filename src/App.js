import React, { Component } from 'react';
import {
  ButtonGroup,
  Col,
  Grid,
  Navbar,
  Panel,
  Row
} from 'react-bootstrap';
import './App.css';

class App extends Component {
  render() {
    return (
      <Grid fluid>
        <Navbar collapseOnSelect fluid>
          <Navbar.Header>
            <Navbar.Brand>
              <a href=".">Delivery Dashboard</a>
              </Navbar.Brand>
          </Navbar.Header>
        </Navbar>
        <Row>
          <Col sm={9}>
            <SearchForm/>
            {/* <CurrentRelease/> */}
          </Col>
          <Col sm={3}>
            <Panel header={<strong>Firefox Releases</strong>}>
              {/* <ReleasesMenu/> */}
            </Panel>
          </Col>
        </Row>
      </Grid>
    );
  }
}

class SearchForm extends Component {
  constructor() {
    super()
    this.state = {value: ""}
  }

  handleChange = (e) => {
    this.setState({value: e.target.value})
  }

  handleClick = () => {
    this.setState({value: ""})
  }

  render() {
    return (
      <form className="search-form well">
        <ClearableTextInput
          onChange={this.handleChange}
          onClick={this.handleClick}
          value={this.state.value}/>
      </form>
    );
  }
}

function ClearableTextInput(props) {
  return (
    <ButtonGroup className="clearable-text">
      <input
        className="form-control"
        onChange={props.onChange}
        placeholder={"Firefox version, eg. \"57.0\""}
        type="search"
        value={props.value}/>
      <span
        className="text-clear-btn"
        onClick={props.onClick}>
        <i className="glyphicon glyphicon-remove"/>
      </span>
    </ButtonGroup>
  )
}

export default App;
