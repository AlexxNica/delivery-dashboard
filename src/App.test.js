import React from 'react';
import {App, parseUrl} from './App';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import createStore from './create-store';

// Mock the Notification API.
global.Notification = {
  requestPermission: jest.fn(),
};

// Mock the Pollbot version (version won't be visible in the rendered
// component, as it's only visible after the state has been updated, not on
// first render.
function fetchMocker(response) {
  return jest
    .fn()
    .mockImplementation(() => Promise.resolve({json: () => response}));
}

global.fetch = fetchMocker({
  version: 'pollbot-version-number',
  commit: 'pollbot-commit-hash',
  source: 'https://github.com/mozilla/PollBot.git',
  name: 'pollbot',
});

// Mock the delivery-dashboard version
jest.mock('./version', () => ({
  version: 'version-number',
  commit: 'commit-hash',
  source: 'https://github.com/mozilla/delivery-dashboard.git',
  name: 'delivery-dashboard',
}));

describe('App', () => {
  it('renders without crashing', () => {
    const app = renderer.create(
      <Provider store={createStore()}>
        <App />
      </Provider>,
    );
    expect(app.toJSON()).toMatchSnapshot();
  });
  it('requests for Notification permissions', () => {
    renderer.create(
      <Provider store={createStore()}>
        <App />
      </Provider>,
    );
    expect(
      global.Notification.requestPermission.mock.calls.length,
    ).toBeGreaterThanOrEqual(1);
  });
  it('requests PollBot for its version', () => {
    renderer.create(
      <Provider store={createStore()}>
        <App />
      </Provider>,
    );
    expect(global.fetch.mock.calls.length).toBeGreaterThanOrEqual(1);
  });
});

describe('parseUrl', () => {
  it('returns null for a non matching url', () => {
    expect(parseUrl('')).toBeNull();
    expect(parseUrl('#foobar')).toBeNull();
  });
  it('returns the proper structure for a matching url', () => {
    expect(parseUrl('#pollbot/firefox/50.0')).toEqual({
      service: 'pollbot',
      product: 'firefox',
      version: '50.0',
    });
  });
});
