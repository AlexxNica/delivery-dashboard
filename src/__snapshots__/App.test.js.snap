// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="."
        >
          Delivery Dashboard
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-9"
      >
        <form
          className="search-form well"
          onSubmit={[Function]}
        >
          <div
            className="clearable-text btn-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Firefox version, eg. \\"57.0\\""
              type="search"
              value=""
            />
            <span
              className="text-clear-btn"
              onClick={[Function]}
            >
              <i
                className="glyphicon glyphicon-remove"
              />
            </span>
          </div>
        </form>
        <p>
          Learn more about a specific version.
          <strong>
             Select or enter your version number.
          </strong>
        </p>
      </div>
      <div
        className="firefox-releases-menu col-sm-3"
      >
        <div
          className="panel panel-default"
          id={undefined}
        >
          <div
            className="panel-heading"
          >
            <strong
              className="panel-title"
            >
              Firefox Releases
            </strong>
          </div>
          <div
            className="panel-body"
          >
            <div
              className="loader"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <p
      className="text-muted"
    >
      Delivery dashboard version:
       
      <a
        href="https://github.com/mozilla/delivery-dashboard/commit/commit-hash"
      >
        version-number
      </a>
       -- Pollbot version:
       
    </p>
  </footer>
</div>
`;
